tez-dag/src/main/java/org/apache/tez/dag/app/DAGAppMaster.java

```diff
@@ -766,6 +768,8 @@ protected synchronized void handle(DAGAppMasterEvent event) {
           "DAGAppMaster Internal Error occurred");
       break;
     case DAG_FINISHED:
+      TezThreadDumpHelper.shutdownPeriodicThreadDumpService(tezThreadDumpHelper);
+      tezThreadDumpHelper = null;
```

Comment:
do we need to invalidate this reference?

Reply:
I did that for safety sake everywhere: if we hold the reference and if a start gets called again, in that case we shouldn’t land up using the old reference but create a new one